index.bs

```diff
@@ -4241,19 +4242,26 @@ promises returned by its <a lt="writable stream writer">writer</a>'s {{WritableS
       },
 
       close() {
-        return new Promise((resolve, reject) => {
-          ws.onclose = resolve;
-          ws.close(1000);
-        });
+        return closeWS(1000);
       },
 
       abort(reason) {
-        return new Promise((resolve, reject) => {
-          ws.onclose = resolve;
-          ws.close(4000, reason && reason.message);
-        });
-      }
+        return closeWS(4000, reason && reason.message);
+      },
     });
+
+    function closeWS(code, reasonString) {
+      return new Promise((resolve, reject) => {
+        ws.onclose = e => {
```

Comment:
The onerror event will fire redundantly. I think it is okay to ignore it completely, so just unsetting the event handler it should work.